src/api.py

```diff
@@ -0,0 +1,62 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+import requests
+import tempfile 
+
+class Response:
+    """
+    this object stores all the information regarding the URL which our API
+    has hit 
+    1) Location of file
+    2) Content type of the file
+    3) Size of the file
+    4) URL of the file
+    """
+    def __init__(self, filename, content_type, size, url):
+        self.filename = filename
+        self.content_type = content_type
+        self.size = size
+        self.url = url
+
+
+def get_request(url):
```

Comment:
What's the purpose of this function that is only a one liner? Is this to help with mock testing? if not it would be best inlined.

Reply:
Actually this was done with the purpose of testing since when I am testing the fetch function with mock object it is not able to write inside the file. That's why I have created two seperate functions instead of testing the api function.

Reply:
Both the functions are testing the whole purpose of the api mechanism, so I think we don't need a seperate test function for api at least for now. Like we have to test that if we fetch a url so are we able to save the data of the url inside our system and that part is totally tested.